src/stcal/jump/jump.py

```diff
@@ -689,50 +549,32 @@ def extend_ellipses(
     for ellipse in ellipses:
         ceny = ellipse[0][0]
         cenx = ellipse[0][1]
-        # Expand the ellipse by the expansion factor. The number of pixels
-        # added to both axes is
-        # the number of pixels added to the minor axis. This prevents very
-        # large flagged ellipses
-        # with high axis ratio ellipses. The major and minor axis are not
-        # always the same index.
-        # Therefore, we have to test to find which is actually the minor axis.
-        if expand_by_ratio:
-            if ellipse[1][1] < ellipse[1][0]:
-                axis1 = ellipse[1][0] + (expansion - 1.0) * ellipse[1][1]
-                axis2 = ellipse[1][1] * expansion
-            else:
-                axis1 = ellipse[1][0] * expansion
-                axis2 = ellipse[1][1] + (expansion - 1.0) * ellipse[1][0]
-        else:
-            axis1 = ellipse[1][0] + expansion
-            axis2 = ellipse[1][1] + expansion
-        axis1 = min(axis1, max_extended_radius)
-        axis2 = min(axis2, max_extended_radius)
+        axes = compute_axes(expand_by_ratio, ellipse, expansion, jump_data)
+
         alpha = ellipse[2]
-        image = cv.ellipse(
-            image,
-            (round(ceny), round(cenx)),
-            (round(axis1 / 2), round(axis2 / 2)),
-            alpha,
-            0,
-            360,
-            (0, 0, jump_flag),
-            -1,
-        )
+        cen = (round(ceny), round(cenx))
+        color = (0, 0, jump_data.fl_jump)
+        image = cv.ellipse(image, cen, axes, alpha, 0, 360, color, -1)
+
         jump_ellipse = image[:, :, 2]
         ngrps = gdq_cube.shape[1]
         last_grp = find_last_grp(grp, ngrps, num_grps_masked)
+
         #  This loop will flag the number of groups
         for flg_grp in range(grp, last_grp):
-            sat_pix = np.bitwise_and(gdq_cube[intg, flg_grp, :, :], sat_flag)
-            saty, satx = np.where(sat_pix == sat_flag)
+            sat_pix = np.bitwise_and(gdq_cube[intg, flg_grp, :, :], jump_data.fl_sat)
+            saty, satx = np.where(sat_pix == jump_data.fl_sat)
             jump_ellipse[saty, satx] = 0
             out_gdq_cube[intg, flg_grp, :, :] = np.bitwise_or(gdq_cube[intg, flg_grp, :, :], jump_ellipse)
     diff_cube = out_gdq_cube - gdq_cube
```

Comment:
```suggestion
```
Expensive, unused calculation.

Reply:
Removed.

Reply:
Looks like it's still there.  This line can actually double the runtime of NIR data with many integrations.

Reply:
I thought I removed it before.  Let's see if I can get it right this time.